packages/pocket-ic/src/lib.rs

```diff
@@ -1282,6 +1284,27 @@ To download the binary, please visit https://github.com/dfinity/pocketic."
 , &bin_path, is_dir, &std::env::current_dir().map(|x| x.display().to_string()).unwrap_or_else(|_| "an unknown directory".to_string()));
     }
 
+    // check PocketIC server version compatibility
+    let mut cmd = Command::new(PathBuf::from(bin_path.clone()));
+    cmd.arg("--version");
+    let version = cmd
+        .output()
+        .unwrap_or_else(|e| {
+            panic!(
+                "Failed to get version of PocketIC binary ({}): {}",
+                bin_path, e
+            )
+        })
+        .stdout;
+    let version_str = String::from_utf8(version)
+        .unwrap_or_else(|e| panic!("Failed to parse PocketIC binary version string: {}", e));
```

Comment:
Should we add a `split` and `strip` and just compare it with the version only, without the name of the binary and the newline at the end?

Reply:
I've stripped the newline at the end, but I'm not sure whether splitting on a space will make the check more robust?

Reply:
I think it would be cleaner, but it's fine :)